### PR TITLE
[osrf_testing_tools_cpp] Add warnings

### PR DESCRIPTION
**Pull request by audrow**
This PR enables `Wformat=2`, `Wconversion`, `Woverloaded-virtual`, `Wshadow`, and `Wnon-virtual-dtor` in `osrf_testing_tools_cpp`.

### CODE REVIEW

File: osrf_testing_tools_cpp/test/memory_tools/test_memory_tools.cpp [hunk 1]
```diff
@@ -200,9 +200,9 @@ TEST(TestMemoryTools, test_example) {
   }, "unexpected malloc");
   // There are also explicit begin/end functions if you need variables to leave the scope
   osrf_testing_tools_cpp::memory_tools::expect_no_malloc_begin();
-  int result = my_second_function(1, 2);
+  int output = my_second_function(1, 2);
   osrf_testing_tools_cpp::memory_tools::expect_no_malloc_end();
-  EXPECT_EQ(result, 3);
+  EXPECT_EQ(output, 3);
```
Comment (clalancette) [inline osrf_testing_tools_cpp/test/memory_tools/test_memory_tools.cpp:205]: I have no particular problem with this change, but why is it needed?

Comment (audrow) [reply osrf_testing_tools_cpp/test/memory_tools/test_memory_tools.cpp:205]: I did it to avoid shadowing the `result` variable in several functions.

Comment (clalancette) [reply osrf_testing_tools_cpp/test/memory_tools/test_memory_tools.cpp:205]: Oh, I missed the shadows down below.  Makes sense, thanks.

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI.

**Comment by audrow** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12527/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7496/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10239/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12446/)

**Comment by audrow** [comment]
Running different gtest versions to see if we can fix the warnings:

* gtest v1.8.0 (what is currently used; not rebuilt from above)
  * macOS [Build Status](http://ci.ros2.org/job/ci_osx/10239/)
* gtest v1.10.0
  * macOS [Build Status](http://ci.ros2.org/job/ci_osx/10352/)

**Comment by audrow** [comment]
Ah, this packages seems to be using its own internal version of gtest (v1.8.1).

**Comment by audrow** [comment]
3 out of 36 warnings remain on macOS after `googletest` was updated. They seem reasonable. I'll try to fix them and then rerun CI.
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10841/)

**Comment by audrow** [comment]
Here is another run of macOS's CI:
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10877/)

**Comment by audrow** [comment]
Here's a full run of CI (I reused macOS CI run from my previous comment):
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13163/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8094/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10877/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13170/)

**Comment by audrow** [comment]
#61 is for the Windows warnings.

**Comment by audrow** [comment]
I've rebased on master now that #61 has been merged. Here's a new CI run:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13196/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8127/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10912/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13213/)

File: osrf_testing_tools_cpp/src/test_runner/main.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ int
 main(int argc, char const * argv[])
 {
   std::vector<std::string> args;
-  args.reserve(argc);
+  args.reserve(static_cast<unsigned long>(argc));
```
Comment (clalancette) [inline osrf_testing_tools_cpp/src/test_runner/main.cpp:42]: According to https://en.cppreference.com/w/cpp/container/vector/reserve, this takes a `size_type`, which for all practical purposes is `size_t`.  So I think we should cast to that instead.

File: osrf_testing_tools_cpp/test/test_runner/assert_env_vars.cpp [hunk 3]
```diff
@@ -91,7 +91,7 @@ int main(int argc, char * argv[])
 {
   ::testing::InitGoogleTest(&argc, argv);
 
-  g_args.reserve(argc);
+  g_args.reserve(static_cast<unsigned long>(argc));
```
Comment (clalancette) [inline osrf_testing_tools_cpp/test/test_runner/assert_env_vars.cpp:94]: Same comment about casting to `size_t` instead.

**Review by clalancette — CHANGES_REQUESTED** [review]

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI.

**Comment by audrow** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13197/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8128/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10913/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13214/)

**Comment by audrow** [comment]
@clalancette, I don't have permission to merge in this repo. Could you merge this in when you have a chance?

**Pull request merged**